diff --git a/pretrain_ectc.py b/pretrain_ectc.py
index 012e7a0..3bcd73c 100644
--- a/pretrain_ectc.py
+++ b/pretrain_ectc.py
@@ -106,13 +106,11 @@ def pretrain(gpu, args):
     
     if args.rank == 0: 
 
-        '''
         wandb.init(config=args, project='ectc')
         wandb.run.name = f"pc-dry-run-{wandb.run.id}"
         wandb.config.update(args)
         config = wandb.config
         wandb.run.save()
-        '''
 
         checkpoint_dir = args.root.joinpath(args.checkpoint_dir)
         if not os.path.exists(checkpoint_dir): 
@@ -184,16 +182,19 @@ def pretrain(gpu, args):
             off_diag = off_diagonal(corr).pow_(2).sum()
             loss = on_diag + args.lambd * off_diag
 
+            wandb.log({"iter loss": loss})
             loss.backwards()
             torch.nn.utils.clip_grad_norm_(pretrainer.parameters(), args.clip)
             optimizer.step()
 
             epoch_loss += loss.item()
             counter += 1
+        wandb.log({"epoch loss": loss})
 
     
+if __name__ == '__main__': 
+    main()
+    wandb.finish()
+
 
 
-    
-    
-    
